Tests/JobsTests/JobsTests.swift

```diff
@@ -414,4 +414,81 @@ final class JobsTests: XCTestCase {
+        }
+        let expectation = XCTestExpectation(description: "TestJob.execute was called", expectedFulfillmentCount: 1)
+        let currentJobTryCount: NIOLockedValueBox<Int> = .init(0)
+        var logger = Logger(label: "JobsTests")
+        logger.logLevel = .trace
+        let jobQueue = JobQueue(
+            .memory,
+            logger: logger
+        )
+        jobQueue.registerJob(
+            parameters: TestParameters.self,
+            retryStrategy: .exponentialJitter(maxAttempts: 3, minJitter: 0.01, maxJitter: 0.25)
+        ) { _, _ in
+            currentJobTryCount.withLockedValue {
+                $0 += 1
+            }
+            expectation.fulfill()
+        }
+        try await testJobQueue(jobQueue) {
+            let cancellableJobId = try await jobQueue.push(TestParameters(), options: .init(delayUntil: .now.addingTimeInterval(1)))
+
+            let isEmpty = await jobQueue.queue.isEmpty()
+            XCTAssertFalse(isEmpty)
+
+            try await jobQueue.cancelJob(jobID: cancellableJobId)
+
+            await fulfillment(of: [expectation], timeout: 5)
+            try await jobQueue.resumeJob(jobID: cancellableJobId)
+            let emptyAfterResume = await jobQueue.queue.isEmpty()
+            XCTAssertTrue(emptyAfterResume)
+        }
+        XCTAssertEqual(currentJobTryCount.withLockedValue { $0 }, 1)
+    }
+
+    func testPausedThenResume() async throws {
+        // Placeholder test since this is the default expectation
+        // other job queue drivers have actual tests which verifies implementation
+        struct TestParameters: JobParameters {
+            static let jobName = "testPausedAndThenResume"
+        }
+        let expectation = XCTestExpectation(description: "TestJob.execute was called", expectedFulfillmentCount: 1)
+        let currentJobTryCount: NIOLockedValueBox<Int> = .init(0)
+        var logger = Logger(label: "JobsTests")
+        logger.logLevel = .trace
+        let jobQueue = JobQueue(
+            .memory,
+            logger: logger
+        )
+        jobQueue.registerJob(
+            parameters: TestParameters.self,
+            retryStrategy: .exponentialJitter(maxAttempts: 3, minJitter: 0.01, maxJitter: 0.25)
+        ) { _, _ in
+            currentJobTryCount.withLockedValue {
+                $0 += 1
+            }
+            expectation.fulfill()
+        }
+        try await testJobQueue(jobQueue) {
+            let pausableJob = try await jobQueue.push(TestParameters(), options: .init(delayUntil: .now.addingTimeInterval(1)))
+            try await jobQueue.pauseJob(jobID: pausableJob)
+
+            let isEmpty = await jobQueue.queue.isEmpty()
+            XCTAssertFalse(isEmpty)
+
+            await fulfillment(of: [expectation], timeout: 5)
+            try await jobQueue.resumeJob(jobID: pausableJob)
+            let emptyAfterResume = await jobQueue.queue.isEmpty()
+            XCTAssertTrue(emptyAfterResume)
```

Comment:
Again I think we should we be testing for when jobs are running and not the state of the internals of the job queue. Then we can use the same test across all queues. With a job queue using one worker, run two jobs, pause the first job, wait for second job to complete, verify first job hasn't run, resume first job and wait for it to run.